intelmq/bots/outputs/templated_smtp/output.py

```diff
@@ -0,0 +1,187 @@
+       text: |
+         {
+           "malware": "{{ event['malware.name'] }}",
+           {%- set comma = joiner(", ") %}
+           {%- for key in event %}
+              {%- if key.startswith('source.') %}
+           {{ comma() }}"{{ key }}": "{{ event[key] }}"
+              {%- endif %}
+           {%- endfor %}
+         }
+       name: report.json
+
+You are responsible for making sure that the text produced by the
+template is valid according to the content-type.
+
+Parameters:
+
+attachments: list of objects with structure:
+              - content-type: string, templated, content-type to use.
+                text: string, templated, attachment text.
+                name: string, templated, filename of attachment.
+
+body: string, templated, body text.
+
+mail_from: string, templated, sender address.
+
+mail_to: string, templated, recipient addresses, comma-separated.
+
+smtp_host: string, hostname of SMTP server.
+
+smtp_password: string, default null, password (if any) for
+               authenticated SMTP.
+
+smtp_port: integer, default 25, TCP port to connect to.
+
+smtp_username: string, default null, username (if any) for
+               authenticated SMTP.
+
+tls: boolean, default false, whether to use use SMTPS. If true, also
+     set smtp_port to the SMTPS port.
+
+starttls: boolean, default true, whether to use opportunistic STARTTLS
+          over SMTP.
+
+subject: string, templated, e-mail subject line.
+
+verify_cert: boolean, default true, whether to verify the server
+             certificate in STARTTLS or SMTPS.
+
+"""
+
+import io
+import smtplib
+import ssl
+from typing import List, Optional
+from email.message import EmailMessage
+try:
+    from jinja2 import Template
+except:
+    Template = None
+
+from intelmq.lib.bot import Bot
+from intelmq.lib.exceptions import ConfigurationError
+from intelmq.lib.exceptions import MissingDependencyError
+
+
+class TemplatedSMTPOutputBot(Bot):
+    smtp_host: Optional[str] = None
+    smtp_port: int = 25
+    ssl: bool = False
+    starttls: bool = False
+    username: Optional[str] = None
+    password: Optional[str] = None
+    verify_cert: bool = True
+    attachments: List[str] = []
+    mail_from: Optional[str] = None
+    mail_to: Optional[str] = None
+    subject: Optional[str] = None
+    body: Optional[str] = None
+
+    def init(self):
+        if not Template:
+            raise MissingDependencyError("jinja2")
+
+        if not self.smtp_host:
+            raise ConfigurationError("Server", "No SMTP host specified")
+        if self.ssl:
+            self.smtp_class = smtplib.SMTP_SSL
+        else:
+            self.smtp_class = smtplib.SMTP
+        if not self.mail_from:
+            raise ConfigurationError("Mail", "No sender specified")
+        if not self.mail_to:
+            raise ConfigurationError("Mail", "No recipient specified")
+        if not self.subject:
+            raise ConfigurationError("Mail", "No subject specified")
```

Comment:
It might make sense to not enforce the subject and the body and simply use sensible default values

Reply:
I'm not sure they're sensible, but there are now default values.